src/experimental/tools/mediaCapabilitiesProber/probers/isTypeSupportedWithFeatures.ts

```diff
@@ -14,15 +14,101 @@
  * limitations under the License.
  */
 
-import isNullOrUndefined from "../../../../../utils/is_null_or_undefined";
-import type { IMediaConfiguration } from "../../types";
-import { findDefaultVideoCodec } from "../defaultCodecsFinder";
+import globalScope from "../../../../utils/global_scope";
+import isNullOrUndefined from "../../../../utils/is_null_or_undefined";
+import type { IMediaConfiguration } from "../types";
+import { findDefaultVideoCodec } from "./defaultCodecsFinder";
+
+type ISupportWithFeatures = "" | "Maybe" | "Not Supported" | "Probably";
+
+type IGlobalScopeWithMSMediaKeysFeatures = typeof globalScope & {
+  MSMediaKeys: {
+    isTypeSupportedWithFeatures: (
+      type: string,
+      features: string | null,
+    ) => ISupportWithFeatures;
+  };
+};
+
+/**
+ * @returns {boolean}
+ */
+function isTypeSupportedWithFeaturesAPIAvailable(): boolean {
+  if (!("MSMediaKeys" in globalScope)) {
+    // MSMediaKeys API not available
+    return false;
+  }
+  if (
+    !(
+      "isTypeSupportedWithFeatures" in
+      (globalScope as IGlobalScopeWithMSMediaKeysFeatures).MSMediaKeys
+    )
+  ) {
+    // isTypeSupportedWithFeatures not available
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Rely on `MSMediaKeys`-only API `isTypeSupportedWithFeatures` to check for
+ * several potential features.
+ * @param {Object} config
+ * @returns {Promise}
+ */
+/* eslint-disable-next-line @typescript-eslint/require-await */
+export default async function probeTypeWithFeatures(
+  config: IMediaConfiguration,
+): Promise<"Supported" | "NotSupported" | "Unknown"> {
+  if (!isTypeSupportedWithFeaturesAPIAvailable()) {
+    throw new Error("MSMediaKeys.isTypeSupportedWithFeatures is not available");
+  }
+  const keySystem = config.keySystem;
+
+  const type = (() => {
+    if (
+      keySystem === undefined ||
+      keySystem.type === undefined ||
+      keySystem.type.length === 0
+    ) {
+      return "org.w3.clearkey";
+    }
+    return keySystem.type;
+  })();
+
+  const features = formatTypeSupportedWithFeaturesConfigForAPI(config);
+
+  const result: ISupportWithFeatures = (
+    globalScope as IGlobalScopeWithMSMediaKeysFeatures
+  ).MSMediaKeys.isTypeSupportedWithFeatures(type, features);
```

Comment:
I'm not sure how this API works, do you have any documentation on this one?

Reply:
It was an old edge feature only present on `MSMediaKeys` and was able to check for more precize policy support.

I also have trouble finding documentation on it today, my searches for now only lead to the RxPlayer and complaints from Netflix users :p

Reply:
TypeScript at some point had it in its type definitions: https://raw.githubusercontent.com/Microsoft/TypeScript/v2.9.2/lib/lib.es6.d.ts

Though not anymore